diesel_cli/src/main.rs

```diff
@@ -186,6 +188,59 @@ fn convert_absolute_path_to_relative(target_path: &Path, mut current_path: &Path
     result.join(target_path.strip_prefix(current_path).unwrap())
 }
 
+fn run_infer_schema(matches: &ArgMatches) {
+    let database_url = database::database_url(matches);
+    let schema_name = None;
+    let schema = diesel_infer_schema::infer_schema_for_schema_name(&database_url, schema_name);
```

Comment:
I assume most of the code in that function and its call stack are basically what there were as part of diesel_codegen? Of not, can you point me to important differences?

Also, we should think about error handling here. The codegen code contains a lot of `unwrap` (and the diff suggests you added some more :)), but I think it won't be too difficult to move it to `Result`. Having done that, I'd really like for the function called here to return an `Iterator<Item=(TableName, Result<Tokens, _>)>` so we can emit as many `table!` definitions as possible and not crash at the first error. This doesn't need to be in the first version though.